packages/driver/src/cypress.js

```diff
@@ -596,6 +598,21 @@ class $Cypress {
     return throwPrivateCommandInterface('addUtilityCommand')
   }
 
+  get currentTest () {
+    const r = this.cy.state('runnable')
+
+    if (!r) {
+      const invocationStack = $stackUtils.getInvocationDetails(this.cy.state('specWindow'), this.config)?.stack
```

Comment:
Might be more straightforward to set `this.specWindow` in `onSpecWindow` instead of using `cy.state`.

Reply:
addressed